### PR TITLE
Initial commit of code for Coursera OAuth2 client, mostly cribbed from courseraprogramming repository

### PR DESCRIPTION
**Pull request by cliu587**
PTAL @saeta 

Changes mainly in the README as compared to courseraprogramming repo. The Python code is all copied with very minor name changes.

Users who wish to use Coursera OAuth2 protected resources other than managing graders will need to set up `~/.coursera/courseraoauth2client.cfg` according to instructions given. We can support a `setup` function to prompt users to enter their client id, client secret, scopes, etc. as well.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+language: python
+python:
+  - "2.7"
+
+# command to install dependencies
+install:
+  - "pip install ."
```
Comment (saeta) [inline .travis.yml:7]: should this be `python setup.py develop`?

File: README.rst [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+-----
+
+Before using Coursera's OAuth2 APIs, be sure you know your client id,
+client secret, and scopes you want for for your application. You may create
+an application at https://accounts.coursera.org/console. When creating the
+application, set the
+``Redirect URI`` to be ``http://localhost:9876``.
+
+Then put your client id, client secret, and scopes
+in a file at ``~/.coursera/courseraoauth2client.cfg``::
```
Comment (saeta) [inline README.rst]: So, the problem with this, is that if you have 2 apps that use this library, they both would end up stomping on each other for this same file. Could you look for maybe a better way to do this? (i.e. load up a file that's included in the package of the scripts that use it?) You'd probably need a fallback for interactive execution tho...

File: README.rst [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+	import requests
+	from courseraoauth2client import oauth2
+	...
+	url = 'https://api.coursera.org/api/externalBasicProfiles.v1?q=me&fields=name'
+	auth = oauth2.build_oauth2().build_authorizer()
+	response = requests.get(url, auth=auth)
+
+Note the first time, you will need to open your browser and authorize your
+Coursera account information. This information will subsequently be saved in
+the ``~/.coursera/oauth2_cache.pickle`` file.
```
Comment (saeta) [inline README.rst]: This will also need to be customized per-application, and not shared.

File: courseraoauth2client/oauth2.py [hunk 4]
```diff
@@ -0,0 +1,504 @@
...
+        scopes=scopes,
+        token_cache_file=cache_filename
+    )
+
+
+def configuration():
+    'Loads configuration from the file system.'
+    defaults = '''
+[oauth2]
+client_id = NS8qaSX18X_Eu0pyNbLsnA
```
Comment (saeta) [inline courseraoauth2client/oauth2.py]: Please remove `client_id`, and `client_secret`, as they do not apply to arbitrary apps. Instead, they must be registered via the app console for each individual app.

File: .travis.yml (hunk 1 above)
Comment (cliu587) [inline .travis.yml:7]: I don't quite follow - this was taken from `courseraprogramming`: https://github.com/coursera/courseraprogramming/blob/master/.travis.yml#L7

What's the difference between using `pip` and `setup.py` for travis-ci?

File: README.rst (hunk 2 above)
Comment (cliu587) [inline README.rst]: Yep. That makes sense. I've tried to change the configuration such that every app that needs to be used is configured separately, and the cache for that app is stored separately. Here's a minimal example of multiple apps using this library (after setting both of them up): https://gist.github.com/cliu587/17a616a4ef2eea13fc2263f1d7d19177

File: courseraoauth2client/oauth2.py (hunk 4 above)
Comment (cliu587) [inline courseraoauth2client/oauth2.py]: Done.

File: README.rst (hunk 3 above)
Comment (cliu587) [inline README.rst]: Done.

File: .travis.yml (hunk 1 above)
Comment (saeta) [inline .travis.yml:7]: ah okay, i'll assume that past-me knew something that present-me has forgotten. :-P looks fine to me.

**Comment by saeta** [comment]
LGTM.

**Pull request merged**